### PR TITLE
fix: new AppleID auth with srp

### PR DESCRIPTION
**Pull request by iowk**
Fixes https://github.com/icloud-photos-downloader/icloud_photos_downloader/issues/970

### CODE REVIEW

File: src/pyicloud_ipd/base.py [hunk 1]
```diff
@@ -190,30 +193,81 @@ def authenticate(self, force_refresh:bool=False, service:Optional[Any]=None) ->
...
-                self.session.post(
-                    "%s/signin" % self.AUTH_ENDPOINT,
+                response = self.session.post(
+                    "%s/signin/complete" % self.AUTH_ENDPOINT,
                     params={"isRememberMeEnabled": "true"},
                     data=json.dumps(data),
                     headers=headers,
                 )
+                if response.status_code == 401:
+                    raise PyiCloudAPIResponseException(response.status_code)
```
Comment (twcurrie) [inline src/pyicloud_ipd/base.py:270]:
```suggestion
                    raise PyiCloudAPIResponseException(response.message , str(response.status_code))
```

`PyiCloudAPIResponseException` takes two [parameters](https://github.com/icloud-photos-downloader/icloud_photos_downloader/blob/8b3adb04eb13cf47c5684530348c7ee347c85fcc/src/pyicloud_ipd/exceptions.py#L13), both must be strings

File: src/pyicloud_ipd/base.py [hunk 2]
```diff
@@ -190,30 +193,81 @@ def authenticate(self, force_refresh:bool=False, service:Optional[Any]=None) ->
...
+            data = {
+                'a': base64.b64encode(A).decode(),
+                'accountName': uname,
+                'protocols': ['s2k', 's2k_fo']
+            }
+
+            try:
+                response = self.session.post("%s/signin/init" % self.AUTH_ENDPOINT, data=json.dumps(data), headers=headers)
+                if response.status_code == 401:
+                    raise PyiCloudAPIResponseException(response.status_code)
```
Comment (twcurrie) [inline src/pyicloud_ipd/base.py:233]:
```suggestion
                    raise PyiCloudAPIResponseException(response.text , str(response.status_code))
```

Similar as below

File: tests/test_authentication.py [hunk 3]
```diff
@@ -126,28 +127,11 @@ def test_2fa_required(self) -> None:
...
-                    "password1",
-                    "--no-progress-bar",
-                    "--cookie-directory",
-                    cookie_dir,
-                    "--auth-only",
-                ],
-                input="0\n654321\n",
-            )
-            assert result.exit_code == 0
+        shutil.copytree(cookie_master_path, cookie_dir)
```
Comment (twcurrie) [inline tests/test_authentication.py:134]: Does this change have to be done in the following files as well?
* [test_listing_libraries.py](https://github.com/icloud-photos-downloader/icloud_photos_downloader/blob/8b3adb04eb13cf47c5684530348c7ee347c85fcc/tests/test_listing_libraries.py#L25-L28)
* [test_listing_albums](https://github.com/icloud-photos-downloader/icloud_photos_downloader/blob/8b3adb04eb13cf47c5684530348c7ee347c85fcc/tests/test_listing_albums.py#L26-L30)

Comment (twcurrie) [reply tests/test_authentication.py:134]: Per [comment](https://github.com/icloud-photos-downloader/icloud_photos_downloader/issues/970#issuecomment-2436796863), this is just a missed change in [PR 971](https://github.com/icloud-photos-downloader/icloud_photos_downloader/pull/971)

Comment (iowk) [reply tests/test_authentication.py:134]: Yes, @AndreyNikiforov mentioned that he will fix it. I will then rebase this branch and the tests should pass.

File: src/pyicloud_ipd/base.py (hunk 1 above)
Comment (iowk) [reply src/pyicloud_ipd/base.py]: Fixed, thank you.

File: src/pyicloud_ipd/base.py (hunk 2 above)
Comment (iowk) [reply src/pyicloud_ipd/base.py]: Fixed.

File: src/pyicloud_ipd/base.py (hunk 1 above)
Comment (twcurrie) [reply src/pyicloud_ipd/base.py]: Thank you for getting this fixed.

File: tests/test_authentication.py (hunk 3 above)
Comment (iowk) [reply tests/test_authentication.py:134]: Rebased. Also fixed CodeQL and type check warnings. Now this PR is ready.

**Pull request merged**